include/seastar/net/stack.hh

```diff
@@ -52,6 +52,8 @@ public:
 
 class socket_impl {
 public:
+    socket_impl() = default;
+    socket_impl(const socket_impl&) = default;
```

Comment:
socket_impl-s are _not_ supposed to be movable and copyable, methinks. These objects are pimpl "backends" for socket-s, and socket-s just keep unique_ptr-s on them. So I'd delete both move- and copy-constructors

(the above statement is not compile-checked nonetheless :) )

Reply:
no, it's supposed to be moveable, as we use `do_with()` on it.

Reply:
but yeah, copy ctor is not necessary and not used. dropped.